tests/integration/test_repository.py

```diff
@@ -1921,6 +1921,26 @@ async def test_service_update_method_no_item_id(author_service: AuthorService, f
     assert updated_obj.name == obj.name
 
 
+async def test_service_update_method_data_is_dict(author_service: AuthorService, first_author_id: Any) -> None:
+    new_date = datetime.date(datetime.now())
+    updated_obj = await maybe_async(
+        author_service.update(item_id=first_author_id, data={"dob": new_date}),
+    )
+    assert updated_obj.dob == new_date
+    # ensure the other fields are not affected
+    assert updated_obj.name == "Agatha Christie"
```

Comment:
This, and the similar assertion in `test_service_update_method_data_is_dict_with_none_value`, fail for the `sync-uuid-mock_sync_engine` and `async-uuid-mock_async_engine`, saying the `name` value is `None` here. This is the case for the existing version of `model_from_dict` as well as this newer version, so it's not a regression. I ran out of time so I couldn't dig further into this for the time being, so I made this a draft PR. Wondering if anyone else has thoughts about this, though.